src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -128,90 +142,104 @@ public void RelayWorker()
         {
             this.logger.LogTrace("()");
 
-            this.asyncLoop = this.asyncLoopFactory.Run($"{this.name}.RelayWorker", async token =>
+            this.relayWorkerTask = Task.Factory.StartNew(async delegate
             {
-                this.logger.LogTrace("()");
-
-                int announceBlockCount = this.blocksToAnnounce.Count;
-                if (announceBlockCount == 0)
+                while (!this.nodeLifetime.ApplicationStopping.IsCancellationRequested)
                 {
-                    this.logger.LogTrace("(-)[NO_BLOCKS]");
-                    return;
-                }
+                    // Wait until a new block is added to the queue.
+                    this.blockEnqueued.Wait();
```

Comment:
you can't do that, it blocks without cancellation

Reply:
Fixed.